src/recon_buildblock/PLSPrior.cxx

```diff
@@ -0,0 +1,663 @@
+                       (*penalty_sptr)[z][y][x]) );
+              }
+              else{
+
+                  (*gradientx_sptr)[z][y][x+1] =
+                          (((*pet_im_grad_x_sptr)[z][y][x+1]-(*anatomical_im_grad_x_sptr)[z][y][x+1]*(*inner_product_sptr)[z][y][x+1]/(*get_norm_sptr ())[z][y][x+1])/
+                          (*penalty_sptr)[z][y][x+1] -
+                          ((*pet_im_grad_x_sptr)[z][y][x]-(*anatomical_im_grad_x_sptr)[z][y][x]*(*inner_product_sptr)[z][y][x]/(*get_norm_sptr ())[z][y][x])/
+                          (*penalty_sptr)[z][y][x]);
+
+                  (*gradienty_sptr)[z][y+1][x] =
+                          (((*pet_im_grad_y_sptr)[z][y+1][x]-(*anatomical_im_grad_y_sptr)[z][y+1][x]*(*inner_product_sptr)[z][y+1][x]/
+                          (*get_norm_sptr ())[z][y+1][x])/(*penalty_sptr)[z][y+1][x] -
+                          (((*pet_im_grad_y_sptr)[z][y][x]-(*anatomical_im_grad_y_sptr)[z][y][x]*(*inner_product_sptr)[z][y][x]/(*get_norm_sptr ())[z][y][x])/(*penalty_sptr)[z][y][x]) );
+
+                  (*gradientz_sptr)[z+1][y][x] =
+                      (((*pet_im_grad_z_sptr)[z+1][y][x]-(*anatomical_im_grad_z_sptr)[z+1][y][x]*(*inner_product_sptr)[z+1][y][x]/
+                          (*get_norm_sptr ())[z][y+1][x])/(*penalty_sptr)[z+1][y][x] -
+                          (((*pet_im_grad_y_sptr)[z][y][x]-(*anatomical_im_grad_y_sptr)[z][y][x]*(*inner_product_sptr)[z][y][x]/(*get_norm_sptr ())[z][y][x])/(*penalty_sptr)[z][y][x]) );
+              }
+              }}}
+
+  for (int z=min_z; z<=max_z; z++)
+    {
+
+      const int min_y = current_image_estimate[z].get_min_index();
+      const int max_y = current_image_estimate[z].get_max_index();
+
+      for (int y=min_y;y<= max_y;y++)
+        {
+
+
+          const int min_x = current_image_estimate[z][y].get_min_index();
+          const int max_x = current_image_estimate[z][y].get_max_index();
+
+          for (int x=min_x;x<= max_x;x++)
+            {
+
+              (*gradient_sptr)[z][y][x] = -((*gradientz_sptr)[z][y][x] + (*gradienty_sptr)[z][y][x] + (*gradientx_sptr)[z][y][x]);
+
+//                        std::cout<<"grad ="<<gradient[z][y][x]<<std::endl;
+
+              if (do_kappa)
+                  (*gradient_sptr)[z][y][x] *=
+                  (*kappa_ptr)[z][y][x] ;
+
+
+
+                        prior_gradient[z][y][x]= (*gradient_sptr)[z][y][x] * this->penalisation_factor;
+            }}}
+
+//write_to_file("pgrad", prior_gradient);
+
+  info(boost::format("Prior gradient max %1%, min %2%\n") % prior_gradient.find_max() % prior_gradient.find_min());
+
+  static int count = 0;
+  ++count;
+  if (gradient_filename_prefix.size()>0)
+    {
+      char *filename = new char[gradient_filename_prefix.size()+100];
+      sprintf(filename, "%s%d.v", gradient_filename_prefix.c_str(), count);
+      //write_to_file(filename, prior_gradient);
```

Comment:
I guess uncomment

Reply:
Also, can you add this to the documentation/release_3.1.htm. Ideally also to the UsersGuide...

Shall I add it at the end of the file?

Reply:
It needs to go into the "New functionality" section of the release notes.

In the Users Guide, it will be wherever the QP is.